R/utility_spliceHelper_functions.R

```diff
@@ -572,6 +588,43 @@ dropGrangesListElementsByWidth <- function(grangesList, minWidth = 5,
     return(relist(unlistedExons, partitioning))
 }
 
+#' Function that selects the first exon from an IRangesList object
+#' @param range IRangesList with elements required to be ordered by coordinates
+#' @param stand strand
+#' @noRd
+selectStartExonFromRangesList <- function(range, strand){
+  lastExons <- as.numeric(cumsum(elementNROWS(range)))
+  startExonsSet <- c(1, lastExons[-(length(lastExons))]+1)
+  startExonsSet[strand == "-"] <- lastExons[strand == "-"]
+  return(unlist(range, use.names = FALSE)[startExonsSet])
+}
+
+#' Function that selects the last exon from an IRangesList object
+#' @param range IRangesList with elements required to be ordered by coordinates
+#' @param stand strand
+#' @noRd
+selectEndExonFromRangesList <- function(range, strand){
+  endExonsSet <- as.numeric(cumsum(elementNROWS(range)))
+  firstExons <- c(1, endExonsSet[-(length(endExonsSet))]+1)
+  endExonsSet[strand == "-"] <- firstExons[strand == "-"]
+  return(unlist(range, use.names = FALSE)[endExonsSet])
+}
+####the ones with relist()
```

Comment:
why are these included here, do we plan to use them later?

Reply:
I have removed them